xpk.py

```diff
@@ -4053,6 +4053,41 @@ def get_workload_list(args) -> tuple[int, str]:
     }
   s = ','.join([key + ':' + value for key, value in columns.items()])
 
+  if args.wait_for_job_completion:
+    # Check that the workload exists
+    args.workload = args.wait_for_job_completion
+    workload_exists = check_if_workload_exists(args)
+    if not workload_exists:
+      xpk_print(
+          f'Workload named {args.workload} not found.'
+      )
+      xpk_exit(1)
+
+    # Get the full workload name
+    get_jobset_name_cmd = f'kubectl get workloads | grep jobset-{args.workload}'
+    return_code, return_value = run_command_for_value(
+      get_jobset_name_cmd, "Get Jobset name", args)
+    lines = return_value.split(' ')
+
+    # Call kubectl wait on the workload using the full workload name
+    timeout_val = args.timeout if args.timeout else -1
+    # pylint: disable=line-too-long
+    wait_cmd = f'kubectl  wait --for jsonpath=\'.status.conditions[-1].type\'=Finished workload {lines[0]} --timeout={timeout_val}s'
+    return_code = run_command_with_updates(
+      wait_cmd, "Wait for workload to finish", args
```

Comment:
nit, might be worth adding the timeout_val}s in the task name so it viewed in the cmd output and can be compared againnst the run time of the task.

ie: f'Wait for workload to finish with timeout: {timveout_val if timveout_val > 0 else 'infinite'} seconds'

Reply:
Done!